### PR TITLE
API Strongly-type action method signatures

### PR DESCRIPTION
**Pull request by emteknetnz**
Issue https://github.com/silverstripe/silverstripe-framework/issues/10466

### CODE REVIEW

File: src/AddToCampaignHandler.php [hunk 1]
```diff
@@ -42,7 +42,8 @@
  *
  * To use, you'd add an action to your LeftAndMain subclass, like this:
  *
- *     function addtocampaign($data, $form) {
+ *     function addtocampaign(array $data, Form $form): HTTPResponse
+ *     {
  *         $handler = AddToCampaignHandler::create($form, $data);
  *         return $handler->handle();
```
Comment (GuySartorelli) [inline src/AddToCampaignHandler.php]: The `handle()` method in this class doesn't always return `HTTPResponse`

File: src/CampaignAdmin.php [hunk 2]
```diff
@@ -730,13 +730,9 @@ public function getCampaignCreateForm()
...
     /**
-     * Save  handler
-     *
-     * @param array $data
-     * @param Form $form
-     * @return HTTPResponse
+     * Save handler
      */
-    public function save($data, $form)
+    public function save(array $data, Form $form): HTTPResponse
```
Comment (GuySartorelli) [inline src/CampaignAdmin.php:735]: `Security::PermissionFailure()` sometimes returns an array

**Review by GuySartorelli — CHANGES_REQUESTED** [review]

File: src/AddToCampaignHandler.php (hunk 1 above)
Comment (emteknetnz) [reply src/AddToCampaignHandler.php]: Reverted

File: src/CampaignAdmin.php (hunk 2 above)
Comment (emteknetnz) [reply src/CampaignAdmin.php:735]: As noted in other PRs, it doesn't

**Review by GuySartorelli — APPROVED** [review]
Not to be merged until the other PRs are ready to avoid breaking CI

**Pull request merged**